sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs

```diff
@@ -237,7 +237,7 @@ await QueueChunk(
         /// Processes the job to job parts
         /// </summary>
         /// <returns>An IEnumerable that contains the job chunks</returns>
-        public abstract Task ProcessPartToChunkAsync();
+        public abstract Task ProcessPartToChunkAsync(CancellationToken cancellationToken = default);
```

Comment:
Odd to me this method is getting a token added to its signature but all the other changes here are using a token saved to this class to pass forwards. I can't tell the distinction from this diff. Can you explain the reasoning?

Reply:
My thought was we need to plumb the CancellationToken that is passed to [ProcessPartAsync()](https://github.com/Azure/azure-sdk-for-net/pull/46712/files#diff-4e409088014d5a400ed06a2939efc3a7070e877d568022f8de3d957278b7e674L110) in the TransferManager. Because if we don't plumb this CancellationToken in the TransferManager, then if the caller cancels the token, we will not be responsive to the cancel.

Reply:
This is related to why I did what I did in [ServiceToServiceJobPart.cs](https://github.com/Azure/azure-sdk-for-net/pull/46712/files/a827679371aca391edb76937637634b83c675a43#diff-09472be2c337d5608a3ca990297f1d40c1cbdbe708c4ce8f2e488c541487ac8d). I needed to link the member token to this passed-in token since the rest of the class uses the member token but it needs to also be responsive to the passed-in token. 

This is def a strange way to do this so let me know if its wrong